benchmark/nixlbench/src/main.cpp

```diff
@@ -80,10 +80,11 @@ static std::vector<std::vector<xferBenchIOV>> createTransferDescLists(xferBenchW
 
         for (const auto &iov : iov_list) {
             for (size_t i = 0; i < count; i++) {
-                size_t offset = ((i * stride) % iov.len);
+                size_t dev_offset = ((i * stride) % iov.len);
 
                 for (size_t j = 0; j < batch_size; j++) {
-                    xfer_list.push_back(xferBenchIOV((iov.addr + offset) + (j * block_size),
+                    size_t block_offset = ((j * block_size) % iov.len);
```

Comment:
Wonder whether these loops should be somehow scaled to have resulting xfer_list.size() to be divided by batch_size while every one of xferBenchIOV in xfer_list to include batch_size number of ops and it all should total to the same amount of read/written memory.

Reply:
Nvm, I see that batch size is taken into account when calculating the stats instead of scaling the amount of transfer requests.